lib/runtime/storage/trie.go

```diff
@@ -194,20 +205,24 @@ func (s *TrieState) DeleteChildLimit(key []byte, limit *[]byte) (uint32, bool, e
 		keys = append(keys, k)
 	}
 	sort.Strings(keys)
-	deleted := uint32(0)
 	for _, k := range keys {
-		tr.Delete([]byte(k))
+		// TODO have a transactional/atomic way to delete multiple keys in trie.
```

Comment:
This seems like a good idea, do we have an issue or discussion opened for this?

Reply:
Good point, created https://github.com/ChainSafe/gossamer/issues/3032 and linked in that comment